hubblestack/extmods/grains/hostuuid.py

```diff
@@ -0,0 +1,28 @@
+# -*- coding: utf-8 -*-
+'''
+Generate a unique uuid for this host, storing it on disk so it persists across
+restarts
+'''
+import logging
+import os
+import uuid
+
+log = logging.get_logger(__name__)
+
+
+def host_uuid():
+    '''
+    Generate a unique uuid for this host, storing it on disk so it persists
+    across restarts
+    '''
+    cached_uuid = os.path.join(os.path.dirname(__opts__['configfile']), 'hubble_cached_uuid')
+    try:
+        if os.path.isfile(cached_uuid):
+            with open(cached_uuid, 'r') as f:
+                return {'host_uuid': f.read()}
```

Comment:
Can the existing host_uuid be compared to the one read from the file, compared against each other, and emit a log message with relevant details if they don't match?

Also, would it be possible to log differently a failure to read a cached host_uuid if there is supposed to be one? This would show that there _was_ a host_uuid but it got removed. Given that host_uuid exists in program memory, but disappeared on disk, we should probably attempt to save it by writing the same one back.

Finally, a failed attempt to write the host_uuid should also be logged.

I think changing host_uuid's should not be easy or expected. All possible avenues of changes to this field should be watched for and logged to establish continuity between the two values.

Reply:
Done. I think I hit on all your points.